File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/ObjC/CRUD/CRUD/ViewControllers/RelatedViewController.m

#import "RelatedViewController.h"
#import "CodeSampleViewController.h"
#import "Utils.h"
#import <Backendless-Swift.h>

#define FIND_RELATED @"Find with related"
#define SHOW_SAMPLE @"ShowCodeSampleFromRelated"

@interface RelatedViewController() {
    NSMutableArray<NSString *> *related;
}
@end

@implementation RelatedViewController

- (void)viewDidLoad {
    [super viewDidLoad];    
    related = [NSMutableArray<NSString *> new];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.fields.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"RelatedCell" forIndexPath:indexPath];
    cell.textLabel.text = [self.fields objectAtIndex:indexPath.row];
    return cell;
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    CodeSampleViewController *codeSampleVC = [segue destinationViewController];
    codeSampleVC.tableName = self.tableName;
    codeSampleVC.operation = FIND_RELATED;
    codeSampleVC.related = related.firstObject;
}

- (IBAction)next:(id)sender {
    if (related.count != 1) {
        Fault *fault = [[Fault alloc] initWithMessage:@"Please select only one related field to load its objects" faultCode:0];
        [utils showErrorAlert:self fault:fault];
    }
    else {
        [self performSegueWithIdentifier:SHOW_SAMPLE sender:nil];
    }
}

- (IBAction)selectProperty:(id)sender {
    if ([[[sender superview] superview] isKindOfClass:[UITableViewCell class]]) {
        UITableViewCell *cell = (UITableViewCell *)[[sender superview] superview];
        NSIndexPath *indexPath = [self.tableView indexPathForCell:cell];
        NSString *property = [self.fields objectAtIndex:indexPath.row];
        if ([(UISwitch *)sender isOn]) {
            [related addObject:property];
        }
        else {
            [related removeObject:property];
        }
    }
}

@end

File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/Swift/CRUD/CRUD/ViewControllers/RelatedViewController.swift

import UIKit
import Backendless

class RelatedViewController: UITableViewController {
    
    var tableName = ""
    var fields = [String]()
    
    private let FIND_RELATED = "Find with related"
    private let SHOW_SAMPLE = "ShowCodeSampleFromRelated"
    
    private let utils = Utils.shared
    
    private var related = [String]()
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return fields.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "RelatedCell", for: indexPath)
        cell.textLabel?.text = fields[indexPath.row]
        return cell
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if let codeSampleVC = segue.destination as? CodeSampleViewController {
            codeSampleVC.tableName = tableName
            codeSampleVC.operation = FIND_RELATED
            codeSampleVC.related = related.first
        }
    }
    
    @IBAction func next(_ sender: Any) {
        if related.count != 1 {
            let fault = Fault(message: "Please select only one related field to load its objects", faultCode: 0)
            utils.showErrorAler(target: self, fault: fault)
        }
        else {
            performSegue(withIdentifier: SHOW_SAMPLE, sender: nil)
        }
    }
    
    @IBAction func selectProperty(_ sender: Any) {
        if let sender = sender as? UISwitch,
            let cell = sender.superview?.superview as? UITableViewCell,
            let indexPath = tableView.indexPath(for: cell) {
            let property = fields[indexPath.row]
            if sender.isOn {
                related.append(property)
            }
            else if let index = related.firstIndex(of: property) {
                related.remove(at: index)
            }
        }
    }
}
